sdk/src/Core/Amazon.Runtime/Credentials/Internal/_bcl45+netstandard/SsoTokenUtils.cs

```diff
@@ -162,5 +179,17 @@ public static SsoToken FromJson(string json)
             
             return token;
         }
+        #region private methods
+        //The string formatted date which comes in RFC3339 format with no milliseconds
+        private static bool ConvertStringToDateTime(string stringFormattedDate, out DateTime dateTime)
+        {
+            dateTime = new DateTime();
+            if (stringFormattedDate == null)
+            {
+                return false;
+            }
+            return DateTime.TryParseExact(stringFormattedDate, AWSSDKUtils.RFC3339NoMsFormat, null, System.Globalization.DateTimeStyles.AssumeUniversal, out dateTime);
```

Comment:
Should this use `XmlConvert.ToDateTime` to give symmetry to the `XmlConvert.ToString` saving code over in CoreAmazonSSOOIDC.cs?

That would match [Sso Token loading](https://github.com/aws/aws-sdk-net/blob/8c6a77b03b0f967388d57e912bc67238f2c7ab19/sdk/src/Core/Amazon.Runtime/Credentials/Internal/_bcl45%2Bnetstandard/SsoTokenUtils.cs#L141)

Reply:
I will see if it's possible to use `XmlConvert.ToDateTime`, if it doesn't complain about formatting then I will use that for better maintainability.

Reply:
After testing, I decided to leave it as `TryParseExact`. `XmlConvert.ToDateTime` is converting it incorrectly, (for example, when the test says it expires in 1 hour, xmlConvert.ToDateTime return something that expires in 5 hours). I think it is because the original `ssoToken.RegistrationExpiresAt` was converted from a timestamp in Unix Epoch time but not completely certain. But to be safe I want to be exact with the parsing and leave it as is.

Reply:
That sounds like some additional "Kind" handling is needed. You're working in a time-zone 4 hours off UTC, right? Add in the additional hour and that explains your 5 hour discrepancy. 

My concern is that the field doesn't have parity with how the other datetime field is de/serialized with this implementation.

Reply:
So the issue was when I made the `XmlConvert.ToDateTime` change in the code , I didn't also use `XmlConvert.ToDateTime` in the test case, causing the mismatch. Once I did that it passed all cases. So in the next revision I will push that out.